clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -1648,9 +1648,13 @@ public static <V> Map<String, V> entriesWithPrefix(Map<String, V> map, String pr
      * @param <V> the type of values stored in the map
      */
     public static <V> Map<String, V> entriesWithPrefix(Map<String, V> map, String prefix, boolean strip) {
+        return entriesWithPrefix(map, prefix, strip, false);
+    }
+
+    public static <V> Map<String, V> entriesWithPrefix(Map<String, V> map, String prefix, boolean strip, boolean allowMatchingLength) {
```

Comment:
Can we add a Javadoc for this? It can be largely identical to the one above, just needs to include an additional description for the `allowMatchingLength` parameter.

Reply:
sure, I have added it